src/imitation/algorithms/adversarial/common.py

```diff
@@ -227,6 +241,8 @@ def __init__(
             assert gen_algo_env is not None
             self.gen_train_timesteps = gen_algo_env.num_envs
             if hasattr(self.gen_algo, "n_steps"):  # on policy
+                # TODO(juan) this looks like a bug; could not find n_steps
```

Comment:
It's defined in `OnPolicyAlgorithm` in SB3, in `__init__`. The tests would break if this was missing.

Reply:
Okay. Then we shouldn't accept as gen_algo any BaseAlgorithm, but only an OnPolicyAlgorithm. This is why mypy was complaining then.

Reply:
This is optional, we have a code-path to handle other algorithms. I don't object to changing the if statement to `isinstance(self.gen_algo, OnPolicyAlgorithm)` although it's less Pythonic (duck-typing).

Reply:
Oh sorry, I changed the type of this in the args. I'll fix this.